tracing-core/src/span.rs

```diff
@@ -181,3 +197,44 @@ impl<'a> Record<'a> {
         self.values.is_empty()
     }
 }
+
+// ===== impl Current =====
+
+impl Current {
+    /// TODO(eliza): docs
```

Comment:
Perhaps in a subsequent PR, but resolving the TODO(eliza): docs would be handy.

Reply:
@davidbarsky this PR was opened as a draft to demonstrate the proposed API changes, with the intention of adding documentation and tests after we agreed on whether or not this was the right approach :) 

I'll add all the necessary docs prior to marking this as ready for review.